src/Raven.Server/Documents/Indexes/Persistence/Lucene/Documents/LuceneDocumentConverterBase.cs

```diff
@@ -38,7 +38,14 @@ public abstract class LuceneDocumentConverterBase : IDisposable
 
         protected LuceneDocumentConverterBase(ICollection<IndexField> fields, bool reduceOutput = false)
         {
-            _fields = fields.ToDictionary(x => x.Name, x => x, StringComparer.OrdinalIgnoreCase);
+            _fields = fields.ToDictionary(x => x.Name, x =>
```

Comment:
Currently we are handling numbers by using quite a naive approach:
https://github.com/ayende/ravendb/blob/v4.0/src/Raven.Server/Documents/Queries/Results/MapReduceQueryResultRetriever.cs#L28-L37

However that can still be a problem for some customers since we always force to send doubles, even though a number is an integer.  I will introduce `__reduced_val` as we do in 3.x